modules/apps/27-interchain-accounts/keeper/relay_test.go

```diff
@@ -574,3 +574,119 @@ func (suite *KeeperTestSuite) fundICAWallet(ctx sdk.Context, portID string, amou
 	suite.Require().NotEmpty(res)
 	suite.Require().NoError(err)
 }
+
+// constructs a send from chainA to chainB on the established channel/connection
+// and sends the same coin back from chainB to chainA.
+func (suite *KeeperTestSuite) TestHandleMsgTransfer() {
```

Comment:
I spent sometime hacking on this and came up with the following. Feel free to test it out yourself and see if this satisfies the test scenario. It passes anyway :) 

```
func (suite *KeeperTestSuite) TestHandleMsgTransfer() {
	suite.SetupTest()

	// create an ICA path and initiate a handshake to register interchain account on chainB
	icaPath := NewICAPath(suite.chainB, suite.chainA)
	suite.coordinator.SetupConnections(icaPath)

	err := suite.SetupICAPath(icaPath, TestOwnerAddress)
	suite.Require().NoError(err)

	interchainAccountAddr, found := suite.chainA.GetSimApp().ICAKeeper.GetInterchainAccountAddress(suite.chainA.GetContext(), icaPath.EndpointA.ChannelConfig.PortID)
	suite.Require().True(found)

	// create a transfer path
	transferPath := ibctesting.NewPath(suite.chainA, suite.chainB)
	transferPath.EndpointA.ChannelConfig.PortID = ibctesting.TransferPort
	transferPath.EndpointB.ChannelConfig.PortID = ibctesting.TransferPort

	suite.coordinator.Setup(transferPath)

	amount, ok := sdk.NewIntFromString("9223372036854775808") // 2^63 (one above int64)
	suite.Require().True(ok)
	coinToSendToB := sdk.NewCoin(sdk.DefaultBondDenom, amount)

	// invoke an ibc transfer send msg from chainA to chainB
	msg := transfertypes.NewMsgTransfer(
		transferPath.EndpointA.ChannelConfig.PortID,
		transferPath.EndpointA.ChannelID,
		coinToSendToB,
		suite.chainA.SenderAccount.GetAddress().String(),
		interchainAccountAddr,
		clienttypes.NewHeight(0, 110),
		0,
	)

	_, err = suite.chainA.SendMsgs(msg)
	suite.Require().NoError(err) // message committed

	fungibleTokenPacket := transfertypes.NewFungibleTokenPacketData(coinToSendToB.Denom, coinToSendToB.Amount.String(), suite.chainA.SenderAccount.GetAddress().String(), interchainAccountAddr)

	packet := channeltypes.NewPacket(
		fungibleTokenPacket.GetBytes(),
		1,
		transferPath.EndpointA.ChannelConfig.PortID,
		transferPath.EndpointA.ChannelID,
		transferPath.EndpointB.ChannelConfig.PortID,
		transferPath.EndpointB.ChannelID,
		clienttypes.NewHeight(0, 110),
		0,
	)

	ack := channeltypes.NewResultAcknowledgement([]byte{byte(1)})
	err = transferPath.RelayPacket(packet, ack.Acknowledgement())
	suite.Require().NoError(err) // relay committed

	// check that voucher exists on chain B
	voucherDenomTrace := transfertypes.ParseDenomTrace(transfertypes.GetPrefixedDenom(packet.GetDestPort(), packet.GetDestChannel(), sdk.DefaultBondDenom))

	icaAcc, err := sdk.AccAddressFromBech32(interchainAccountAddr)
	suite.Require().NoError(err)

	// assert the balances are equal
	balance := suite.chainB.GetSimApp().BankKeeper.GetBalance(suite.chainB.GetContext(), icaAcc, voucherDenomTrace.IBCDenom())
	coinSentFromAToB := transfertypes.GetTransferCoin(transferPath.EndpointB.ChannelConfig.PortID, transferPath.EndpointB.ChannelID, sdk.DefaultBondDenom, amount)
	suite.Require().Equal(coinSentFromAToB, balance)

	// build the expected packet and call OnRecvPacket directly on chainA's transferKeeper
        // this bypasses the sigverify from calling a new MsgTransfer on chainB with the interchain account as signer
	fungibleTokenPacket = transfertypes.NewFungibleTokenPacketData(
		transfertypes.GetPrefixedDenom(packet.GetDestPort(), packet.GetDestChannel(), sdk.DefaultBondDenom),
		coinSentFromAToB.Amount.String(),
		interchainAccountAddr,
		suite.chainA.SenderAccount.GetAddress().String(),
	)

	packet = channeltypes.NewPacket(
		fungibleTokenPacket.GetBytes(),
		suite.chainB.SenderAccount.GetSequence(),
		transferPath.EndpointB.ChannelConfig.PortID,
		transferPath.EndpointB.ChannelID,
		transferPath.EndpointA.ChannelConfig.PortID,
		transferPath.EndpointA.ChannelID,
		clienttypes.NewHeight(0, 110),
		0,
	)

	// retrieve the balance before packet processing
	balanceBefore := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), suite.chainA.SenderAccount.GetAddress(), sdk.DefaultBondDenom)

	err = suite.chainA.GetSimApp().TransferKeeper.OnRecvPacket(suite.chainA.GetContext(), packet, fungibleTokenPacket)
	suite.Require().NoError(err)

	// assert the balance after is equal to the balance before + the original sent amount
	balanceAfter := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), suite.chainA.SenderAccount.GetAddress(), sdk.DefaultBondDenom)
	suite.Require().Equal(balanceBefore.Add(coinToSendToB), balanceAfter)
}
```